relentless/potential/potential.py

```diff
@@ -34,28 +131,123 @@ def evaluate(self, pair):
                 params[p] = v
             elif v is not None:
                 raise TypeError('Parameter type unrecognized')
-
             if v is None:
                 raise ValueError('Parameter {} is not set for ({},{}).'.format(p,pair[0],pair[1]))
 
         return params
 
     def save(self, filename):
-        all_params = {}
+        """Saves the data from a coefficient matrix to a file.
+
+        Serializes each parameter value in order to track the internal values
+        of the :py:class:`Variable` object and distinguish it from 'scalar' values.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file to save data in.
+
+        """
+        data = {}
         for key in self:
-            all_params[str(key)] = self.evaluate(key)
+            all_params = {}
+            for param in self[key]:
+                var = self[key][param]
+                if isinstance(var, Variable):
+                    all_params[param] = {'type':'variable', 'value':var.value,
+                                         'const':var.const, 'low':var.low, 'high':var.high}
+                elif isinstance(var, (float,int)):
+                    all_params[param] = {'type':'scalar', 'value':var}
```

Comment:
There should be an else branch here that raises an error if the type is unknown, as in `evaluate`.